### PR TITLE
Fixes Add `n_iter_` attribute after fit to GraphMatch #342

### PR DESCRIPTION
**Pull request by dtborders**
#### Reference Issues/PRs
Fixes issue #342 (Add `n_iter_` attribute after fit to GraphMatch)

#### What does this implement/fix? Explain your changes.
Implements n_iter_ in the same manner as sklearn.cluster.KMeans¶ (https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/cluster/_kmeans.py#L749)

### CODE REVIEW

**Comment by dtborders** [comment]
@bdpedigo I think this is complete, but I hardly had to modify anything. Do you want the default n_iter_ to be zero? SKLearn K Means doesn't have a default value.

**Comment by bdpedigo** [comment]
no default necessary. @asaadeldin11 will give you a review at some point

File: graspy/match/gmp.py
Comment (asaadeldin11) [inline graspy/match/gmp.py:299]: bit of a nit pick but I would prefer `n_iter_best` rather than `best_n_iter`

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]
Please add `n_iter` and a description to the documentation under attributes. See [here](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/cluster/_kmeans.py#L861) for reference. Make sure to note that the `n_iter` corresponds to the initialization returned.

File: graspy/match/gmp.py
Comment (dtborders) [reply graspy/match/gmp.py:299]: Will do, I can change that. Sklearn used best_n_iter: 
Screen Shot 2020-09-12 at 9 44 46 AM
Would you still like me to change it

Comment (asaadeldin11) [reply graspy/match/gmp.py:299]: You can leave as is then.

File: graspy/match/gmp.py
Comment (asaadeldin11) [inline graspy/match/gmp.py:174]: No need to list the attributes here; you can delete this.

File: graspy/match/gmp.py
Comment (asaadeldin11) [inline graspy/match/gmp.py:83]:
```suggestion
        Number of Frank-Wolfe iterations run.
```
Add the following sentence as well: "If `n_init` > 1, `n_iter_` refelects the number of iterations performed at the initialization returned."

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]

File: graspy/match/gmp.py
Comment (dtborders) [reply graspy/match/gmp.py]: updated!

File: graspy/match/gmp.py
Comment (dtborders) [reply graspy/match/gmp.py]: Resolved!

**Comment by dtborders** [comment]
I think this is all ready to go. Just waiting on the deploy-preview

**Comment by asaadeldin11** [comment]
@dtborders you can pretty much consider this approved for the time being; I don't think netlify is working right now, so I'll wait until it is working to officially approve.

**Comment by asaadeldin11** [comment]
@dtborders can you make some inconsequential commit and then revert it to trigger netlify?

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]
Formatting errors showing in Travis. Run black on `graspy/match/gmp.py`.

File: graspy/match/gmp.py
Comment (asaadeldin11) [inline graspy/match/gmp.py:83]:
```suggestion
        Number of Frank-Wolfe iterations run. If n_init > 1, n_iter_ reflects the number of
```

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]

File: graspy/match/gmp.py
Comment (asaadeldin11) [inline graspy/match/gmp.py:83]:
```suggestion
        Number of Frank-Wolfe iterations run. If `n_init` > 1, `n_iter_` reflects the number of
```

**Review by asaadeldin11 — APPROVED** [review]
This looks good to me

**Review by bdpedigo — APPROVED** [review]

**Pull request merged**
